config/main.py

```diff
@@ -1227,6 +1227,91 @@ def validate_patch(patch):
     except Exception as e:
         raise GenericConfigUpdaterError(f"Validate json patch: {patch} failed due to:{e}")
 
+
+def multiasic_validate_single_file(filename):
+    ns_list = [DEFAULT_NAMESPACE, *multi_asic.get_namespace_list()]
+    file_input = read_json_file(filename)
+    file_ns_list = [DEFAULT_NAMESPACE if key == HOST_NAMESPACE else key for key in file_input]
+    if set(ns_list) != set(file_ns_list):
+        click.echo("Input file {} must contain all asics config".format(filename))
+        raise click.Abort()
+
+
+def load_sysinfo_if_missing(asic_config):
+    platform = asic_config.get("DEVICE_METADATA", {}).\
+        get(HOST_NAMESPACE, {}).get("platform")
+    mac = asic_config.get("DEVICE_METADATA", {}).\
+        get(HOST_NAMESPACE, {}).get("mac")
+    if not platform or not mac:
+        log.log_warning("Input file does't have platform or mac. platform: {}, mac: {}"
+                        .format(None if platform is None else platform, None if mac is None else mac))
+        return True
+    return False
+
+
+def flush_configdb(namespace=DEFAULT_NAMESPACE):
+    if namespace is DEFAULT_NAMESPACE:
+        config_db = ConfigDBConnector()
+    else:
+        config_db = ConfigDBConnector(use_unix_socket_path=True, namespace=namespace)
+
+    config_db.connect()
+    client = config_db.get_redis_client(config_db.CONFIG_DB)
+    client.flushdb()
+    return client, config_db
+
+
+def migrate_db_to_lastest(namespace=DEFAULT_NAMESPACE):
+    # Migrate DB contents to latest version
+    db_migrator = '/usr/local/bin/db_migrator.py'
+    if os.path.isfile(db_migrator) and os.access(db_migrator, os.X_OK):
+        if namespace is DEFAULT_NAMESPACE:
+            command = [db_migrator, '-o', 'migrate']
+        else:
+            command = [db_migrator, '-o', 'migrate', '-n', str(namespace)]
+        clicommon.run_command(command, display_cmd=True)
+
+
+def multiasic_write_to_db(filename, load_sysinfo):
+    file_input = read_json_file(filename)
+    for ns in [DEFAULT_NAMESPACE, *multi_asic.get_namespace_list()]:
+        asic_name = HOST_NAMESPACE if ns == DEFAULT_NAMESPACE else ns
+        asic_config = file_input[asic_name]
+
+        asic_load_sysinfo = True if load_sysinfo else False
+        if not asic_load_sysinfo:
+            asic_load_sysinfo = load_sysinfo_if_missing(asic_config)
+
+        if asic_load_sysinfo:
+            cfg_hwsku = asic_config.get("DEVICE_METADATA", {}).\
+                get(HOST_NAMESPACE, {}).get("hwsku")
+            if not cfg_hwsku:
+                click.secho("Could not get the HWSKU from config file,  Exiting!!!", fg='magenta')
+                sys.exit(1)
+
+        client, _ = flush_configdb(ns)
+
+        if asic_load_sysinfo:
```

Comment:
>asic_load_sysinfo [](http://example.com/codeflow?start=11&length=17)

IMHO, this variable equals True that means something missed? Then I am quite confusion we write to db under missing circumstance?

Reply:
No matter the config source is minigraph or Golden Config, the config source itself could not get the mac in advance. The platform info and mac address was generated at run time.
Take single asic as example, Golden Config cannot provide such infomation. But SONiC container will not successfully get up if platform and mac are missing in CONFIGDB. That's why we need to check if sysinfo is missing and generate this info as needed.